Sample/SwiftUI/CustomEventView.swift

```diff
@@ -0,0 +1,132 @@
+                            ActionSheet(
+                                title: Text("Select attribute type"),
+                                buttons: Array(AttributeTypeKeys.allCases
+                                    .map({ type in
+                                        if type == .none {
+                                            return ActionSheet.Button.cancel(Text("Cancel"))
+                                        }
+                                        return ActionSheet.Button.default(Text(type.rawValue)) {
+                                            attributes[editingTextFieldIndex].type = type.rawValue
+                                        }
+                                    }))
+                            )
+                        }
+                    }.textFieldStyle(.roundedBorder)
+                }.padding(.horizontal, 10)
+            }
+            HStack {
+                Button("Add") {
+                    attributes.append(
+                        EventAttribute())
+                }.padding()
+                Spacer()
+            }
+            Button("SEND") {
+                sendEvent()
+            }.padding()
+            Spacer().frame(height: 400)
+        }
+        .alert(isPresented: $isErrorAlertPresented, content: {
+            Alert(title: Text("Invalid input format"))
+        }).onDisappear {
+            /// re-init to reset attribute content and filled with one empty attribute for UI purpose
+            attributes = [
+                EventAttribute()
+            ]
+            eventName = ""
+        }
+    }
+
+    private func sendEvent() {
+        var eventAttributes = [CustomAttribute]()
+        for attribute in self.attributes {
+            guard let customAttribute = EventHelper.customAttributeFromData(name: attribute.name,
+                                                                            value: attribute.value,
+                                                                            type: attribute.type) else {
+                isErrorAlertPresented = true
+                return
+            }
+            eventAttributes.append(customAttribute)
+        }
+
+        guard !eventName.isEmpty else {
+            isErrorAlertPresented = true
+            return
+        }
+
+        RInAppMessaging.logEvent(
+            CustomEvent(withName: eventName,
+                        withCustomAttributes: eventAttributes)
+        )
+    }
+}
+
+@available(iOS 13.0, *)
+struct CustomEventView_Previews: PreviewProvider {
+    static var previews: some View {
+        CustomEventView()
+    }
+}
+
+private struct EventAttribute: Hashable, Identifiable {
+    init(name: String = "", value: String = "", type: String = "") {
+        self.name = name
+        self.value = value
+        self.type = type
+    }
+
+    let id = UUID()
+    var name: String
+    var value: String
+    var type: String
```

Comment:
@ts-robih-robihamanto Can you move this parameters before init function?

Reply:
Updated here [b3a87fc](https://github.com/rakutentech/ios-inappmessaging/pull/278/commits/b3a87fc8fee72033dd8340384020d67b44568791).